library/src/main/java/com/datatorrent/lib/parser/XmlParser.java

```
@@ -0,0 +1,139 @@
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.transform.stream.StreamSource;
+import javax.xml.validation.Schema;
+import javax.xml.validation.SchemaFactory;
+import javax.xml.validation.Validator;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.Document;
+import org.xml.sax.SAXException;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+
+import com.datatorrent.api.DefaultOutputPort;
+import com.datatorrent.netlet.util.DTThrowable;
+
+public class XmlParser extends Parser<String>
+{
+  private String schemaXSDFile;
+  private transient Unmarshaller unmarshaller;
+  private transient Validator validator;
+
+  public transient DefaultOutputPort<Document> parsedOutput = new DefaultOutputPort<Document>();
+
+  @Override
+  public Object convert(String tuple)
+  {
+    // This method is not invoked for XML parser
+    return null;
+  }
+
+  @Override
+  public void processTuple(String inputTuple)
+  {
+    try {
+      if (out.isConnected()) {
+        StringReader reader = new StringReader(inputTuple);
+        JAXBElement<?> output = unmarshaller.unmarshal(new StreamSource(reader), getClazz());
+        LOG.debug(output.getValue().toString());
+        emittedObjectCount++;
+        out.emit(output.getValue());
+      } else {
+        validator.validate(new StreamSource(inputTuple));
+      }
+      if (validatedOutput.isConnected()) {
+        validatedOutput.emit(inputTuple);
+      }
+      if (parsedOutput.isConnected()) {
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder builder;
+        try {
+          builder = factory.newDocumentBuilder();
+          Document doc = builder.parse(inputTuple);
+          parsedOutput.emit(doc);
+
+        } catch (Exception e) {
+          e.printStackTrace();
```

swallowing exception?